trace-resources/src/Cardano/Logging/Resources/os-support-darwin.c

```diff
@@ -0,0 +1,272 @@
+
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <mach/mach.h>
+#include <mach/task.h>
+#include <mach/task_info.h>
+#include <mach/mach_init.h>
+#include <mach/mach_host.h>
+#include <net/route.h>
+#include <net/if_dl.h>
+
+/*
+// includes for c_get_sys_disk_io_counters
+// will require GHC options in cabal file:
+//   if os(darwin)
+//     ghc-options:       -framework CoreFoundation -framework IOKit
+//
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+#include <IOKit/storage/IOBlockStorageDriver.h>
+#include <IOKit/storage/IOMedia.h>
+#include <IOKit/IOBSD.h>
+#include <IOKit/ps/IOPowerSources.h>
+#include <IOKit/ps/IOPSKeys.h>
+*/
+#include "os-support-darwin.h"
+
+
+/* c_get_process_memory_info */
+
+int c_get_process_memory_info(struct mach_task_basic_info *counters, int pid)
+{
+    task_t task = MACH_PORT_NULL;
+    if (task_for_pid(current_task(), pid, &task) != KERN_SUCCESS) {
+        return -2;
+    }
+    struct mach_task_basic_info t_info;
+    mach_msg_type_number_t t_info_count = MACH_TASK_BASIC_INFO_COUNT;
+    if (task_info(task, MACH_TASK_BASIC_INFO, (task_info_t)counters, &t_info_count) != KERN_SUCCESS) {
+        return -1;
+    }
+    return 1;
```

Comment:
In the unix it is more common to return `0` on success.  Then `throwIfNeg` from base can be used directly.

Reply:
See above

Reply:
Ok